core/ordering/cosipbft/pbft/mod.go

```diff
@@ -53,6 +55,29 @@ func (s State) String() string {
 	}
 }
 
+// defines prometheus metrics
+var (
+	promBlocks = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: "dela_cosipbft_blocks_total",
+		Help: "total number of blocks",
+	})
```

Comment:
This looks like it should be a counter, rather than a gauge.
We only expect it to increase, right ?

Reply:
True, but we can't easily initialize it if there are already some blocks. The counter doesn't let you set a value.

Reply:
It depends what we want to measure with this: total blocks or rate of adding new blocks ? 

A counter increments over the lifecycle of the application, which means we can always look at the rate (i.e. blocks/s).
But you're right that if the idea is to monitor the total, then a gauge would be better.